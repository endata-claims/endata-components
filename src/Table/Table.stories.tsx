import React from 'react'
import Table from '.'

export default {
  title: 'Table',
}

const data = [
  { id: 1, email: 'beatyshot@gmail.com', name: 'Yuki', age: 25 },
  { id: 2, email: 'beatyneko@gmail.com', name: 'Yuki', age: 25 },
  { id: 3, email: 'beatyblood@gmail.com', name: 'Yuki', age: 25 },
]

const columns = [
  { Header: 'ID', accessor: 'id' },
  { Header: 'Email', accessor: 'email' },
  { Header: 'Name', accessor: 'name' },
  { Header: 'Age', accessor: 'age' },
]

export const Default = () => {
  return (
    <Table
      loading={false}
      columns={columns}
      data={data}
    />
  )
}

export const Loading = () => {
  return (
    <Table
      loading={true}
      columns={undefined}
      data={undefined}
    />
  )
}

export const LoadingMore = () => {
  return (
    <Table
      loadingMore={true}
      columns={columns}
      data={data}
    />
  )
}