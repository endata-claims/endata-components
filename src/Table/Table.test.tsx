import React from 'react'
import { shallow } from 'enzyme'

import Table from '.'

const data = [
  { id: 1, email: 'beatyshot@gmail.com', name: 'Yuki', age: 25 },
  { id: 2, email: 'beatyneko@gmail.com', name: 'Yuki', age: 25 },
  { id: 3, email: 'beatyblood@gmail.com', name: 'Yuki', age: 25 },
]

const columns = [
  { Header: 'ID', accessor: 'id' },
  { Header: 'Email', accessor: 'email' },
  { Header: 'Name', accessor: 'name' },
  { Header: 'Age', accessor: 'age' },
]

describe('Paper tests', () => {
  const renderWrapper = (props: any) => shallow(<Table data={data} columns={columns} />)

  describe('snapshots', () => {
    it('should render without crash', () => {
      expect(renderWrapper({})).toMatchSnapshot()
    })
    it('should render loading state', () => {
      expect(renderWrapper({ loading: true })).toMatchSnapshot()
    })
    it('should render loading more state', () => {
      expect(renderWrapper({ loadingMore: true })).toMatchSnapshot()
    })
  })
})