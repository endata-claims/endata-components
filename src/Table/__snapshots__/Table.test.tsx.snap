// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper tests snapshots should render loading more state 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-container-1"
>
  <Table
    columns={
      Array [
        Object {
          "Header": "ID",
          "accessor": "id",
        },
        Object {
          "Header": "Email",
          "accessor": "email",
        },
        Object {
          "Header": "Name",
          "accessor": "name",
        },
        Object {
          "Header": "Age",
          "accessor": "age",
        },
      ]
    }
    data={
      Array [
        Object {
          "age": 25,
          "email": "beatyshot@gmail.com",
          "id": 1,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyneko@gmail.com",
          "id": 2,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyblood@gmail.com",
          "id": 3,
          "name": "Yuki",
        },
      ]
    }
  />
</WithStyles(ForwardRef(Paper))>
`;

exports[`Paper tests snapshots should render loading state 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-container-1"
>
  <Table
    columns={
      Array [
        Object {
          "Header": "ID",
          "accessor": "id",
        },
        Object {
          "Header": "Email",
          "accessor": "email",
        },
        Object {
          "Header": "Name",
          "accessor": "name",
        },
        Object {
          "Header": "Age",
          "accessor": "age",
        },
      ]
    }
    data={
      Array [
        Object {
          "age": 25,
          "email": "beatyshot@gmail.com",
          "id": 1,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyneko@gmail.com",
          "id": 2,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyblood@gmail.com",
          "id": 3,
          "name": "Yuki",
        },
      ]
    }
  />
</WithStyles(ForwardRef(Paper))>
`;

exports[`Paper tests snapshots should render without crash 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-container-1"
>
  <Table
    columns={
      Array [
        Object {
          "Header": "ID",
          "accessor": "id",
        },
        Object {
          "Header": "Email",
          "accessor": "email",
        },
        Object {
          "Header": "Name",
          "accessor": "name",
        },
        Object {
          "Header": "Age",
          "accessor": "age",
        },
      ]
    }
    data={
      Array [
        Object {
          "age": 25,
          "email": "beatyshot@gmail.com",
          "id": 1,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyneko@gmail.com",
          "id": 2,
          "name": "Yuki",
        },
        Object {
          "age": 25,
          "email": "beatyblood@gmail.com",
          "id": 3,
          "name": "Yuki",
        },
      ]
    }
  />
</WithStyles(ForwardRef(Paper))>
`;
